File: Day17/maximumLength.java
public class Solution {
    public int maximumLength(int[] nums, int k) {
        int[][] dp = new int[k][k];
        int maxSubsequence = 0;

        for (int num : nums) {
            num = num % k;
            for (int prev = 0; prev < k; prev++) {
                dp[prev][num] = dp[num][prev] + 1;
                maxSubsequence = Math.max(maxSubsequence, dp[prev][num]);
            }
        }

        return maxSubsequence;
    }
}

File: Day17/maximumLength.py
class Solution:
    def maximumLength(self, nums: List[int], k: int) -> int:
        dp = [[0] * k for _ in range(k)]
        maxSubsequence = 0
        for num in nums:
            num %= k
            for prev in range(k):
                dp[prev][num] = dp[num][prev] + 1
                maxSubsequence = max(maxSubsequence, dp[prev][num])
        return maxSubsequence
